### ISSUE TITLE
Refresh docs to highlight assignment and double equals over unification

### ISSUE THREAD

**Issue by tsandall** [open-policy-agent/opa#952]
We have talked for a while about refreshing the documentation to make use of `:=` and `==` instead of `=`. This is more intuitive for new users and avoids potential pitfalls with globals being captured.

Ref:
- #950

**Comment by tim-styra** [comment]
There was a request to put content similar to the following from slack into the description of `:=`.

`x := 7` declares that `x` is a variable local to the rule.

Moreover, the compiler will stop you from writing two such assignments in the same rule, e.g. `x := 7; x := 8` creates a compile-time error.

Neither of those are true with `x = 7`.  `x = 7` will assign `x` to 7 if `x` is unbound, and it will compare `x` to 7 if `x` is already bound.

Moreover, if there is a global variable named `x`, `x = 7` will compare that global value of `x` to 7.

In short, use `:=` if you want assignment inside a rule.  Much clearer for people reading the rule, and gives you what you want.

**Comment by tsandall** [comment]
Fixed by #1259
